### ISSUE TITLE
🐛 BUG: Migration flow for Wrangler1 webpack projects could be more clear

### ISSUE THREAD

**Issue by jkup** [cloudflare/wrangler2#701]
### What version of `Wrangler` are you using?

0.0.0-52ea60f

### What operating system are you using?

Mac

### Describe the Bug

Feel free to close this!

I was messing around with some of the older templates today and tried an old React template (https://github.com/cloudflare/react-workers-template).

I got a bit turned around at the workflow, mainly that `type="webpack"` *might* mean you need a custom build but it's not super clear how you would know. My workflow was

1. Run `wrangler dev`
2. Get an error about type being deprecated
3. Delete type, zone_id and add main field
4. Run `wrangler dev`

I got

```sh
☁  react-workers-template [master] ⚡  wrangler dev
 ⛅️ wrangler 0.0.0-52ea60f
---------------------------
✘ [ERROR] The JSX syntax extension is not currently enabled

    src/index.js:7:16:
      7 │ ReactDOM.render(<App />, document.getElementById('root'));
        ╵                 ^

  The esbuild loader for this file is currently set to "js" but it must be set to "jsx" to be able
  to parse JSX syntax. You can use "loader: { '.js': 'jsx' }" to do that.

Build failed with 1 error:
src/index.js:7:16: ERROR: The JSX syntax extension is not currently enabled
```

So I guess my questions are:

1. Should we detect when type is webpack and tell the user they might need a custom build? Or even direct them to a doc page where we can list when they would or would not want a custom build?
2. Should we try to intercept esbuild errors? I worry that a user might be confused because they wouldn't even know they are using esbuild!

Basically just wondering if we can improve this workflow and, if so, should it happen via CLI responses or just a docs page!

Cheers!

**Comment by threepointone** [comment]
We definitely need a first class webpack migration story, tracking here https://github.com/cloudflare/wrangler2/issues/8

The specific error you got tho, is a proper bug. We have the loader setup correctly here https://github.com/cloudflare/wrangler2/blob/53490df173c7c6439062c48ae7cb2ae0130eb909/packages/wrangler/src/bundle.ts#L61, but not when we try to guess the worker format here https://github.com/cloudflare/wrangler2/blob/53490df173c7c6439062c48ae7cb2ae0130eb909/packages/wrangler/src/entry.ts#L93 

We can definitely work on intercepting esbuild errors, but it's not a straightforward thing. A broader thing to do is fix the templates of course. 

(also, you made a mistake, you should use workers-site/index.js as an entry point, not src/index.js which is the browser side entry point. But it caught the bug! which is good.)

**Comment by threepointone** [comment]
Fix for the error in https://github.com/cloudflare/wrangler2/pull/702

**Comment by threepointone** [comment]
tracking the webpack migration guide in #8, closing this issue
